staging/src/k8s.io/client-go/tools/remotecommand/websocket.go

```diff
@@ -207,25 +222,25 @@ func (c *wsStreamCreator) CreateStream(headers http.Header) (httpstream.Stream,
 	}
 	reader, writer := io.Pipe()
 	s := &stream{
-		headers:   headers,
-		readPipe:  reader,
-		writePipe: writer,
-		conn:      c.conn,
-		connMu:    &c.connMu,
-		id:        id,
+		headers:       headers,
+		readPipe:      reader,
+		writePipe:     writer,
+		conn:          c.conn,
+		connWriteLock: &c.connWriteLock,
+		id:            id,
 	}
 	c.setStream(id, s)
 	return s, nil
 }
 
-// readDemuxLoop is the reading processor for this endpoint of the websocket
+// readDemuxLoop is the lock-free reading processor for this endpoint of the websocket
```

Comment:
What do you mean by lock-free?. Consumer is responsible from thread safety or this function is builtin thread safe?

Reply:
This function is only run in one goroutine, so there is only the one goroutine reading from the websocket. There is no need for a mutex for reading the websocket connnection--only for writing the connection.